src/utils.c

```diff
@@ -7,6 +7,19 @@
 #include <stdlib.h>
 #include <string.h>
 
+char* string_cut_maxlen(char* string, int maxlen){
+        char *buffer;
+
+        if (maxlen < 0 || strlen(string) < maxlen) {
+                buffer = string;
+        } else {
+                size_t buffer_maxlen = maxlen;
+                buffer = calloc(buffer_maxlen, sizeof(char));
```

Comment:
Thanks for pointing this out. My C-Skills are quite limited and I have not known the differences of all available allocation-functions yet.